src/core/lib/gpr/assert_internal.h

```diff
@@ -0,0 +1,42 @@
+// Copyright 2022 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef GRPC_CORE_LIB_GPR_ASSERT_INTERNAL_H
+#define GRPC_CORE_LIB_GPR_ASSERT_INTERNAL_H
+
+#include <grpc/support/port_platform.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+
+/** abort() the process if x is zero, with rudimentary logging to prevent
+   circular dependencies with gpr_log.
+
+   Intended for internal invariants.  If the error can be recovered from,
+   without the possibility of corruption, or might best be reflected via
+   an exception in a higher-level language, consider returning error code.  */
+#define GPR_ASSERT_INTERNAL(x)                     \
+  do {                                             \
+    if (GPR_UNLIKELY(!(x))) {                      \
+      fprintf(stderr, "assertion failed: %s", #x); \
```

Comment:
Might be a good idea to have this log the file and line number.

Reply:
Alright, we now have files & line numbers, and the spacing is consistent with normal logging. To keep the dependencies for this internal target low, I haven't attempted to add in time, date, or tid in a cross-platform way.

```
D0830 02:17:02.298932360      16 test_config.cc:138]         test slowdown factor: sanitizer=1, fixture=1, poller=1, total=1
INTERNAL                         time_posix.cc:69]           DO NOT SUBMIT: 42
I0830 02:17:02.299245791      16 ev_epoll1_linux.cc:121]     grpc epoll fd: 3
D0830 02:17:02.299266219      16 ev_posix.cc:141]            Using polling engine: epoll1
D0830 02:17:02.299306255      16 dns_resolver_ares.cc:831]   Using ares dns resolver
```